raiden/network/proxies/token_network.py

```diff
@@ -194,26 +205,65 @@ def _new_netting_channel(self, partner: typing.Address, settle_timeout: int):
 
         return transaction_hash
 
-    def channel_exists(self, participant1: typing.Address, participant2: typing.Address) -> bool:
-        channel_data = self.detail_channel(participant1, participant2)
+    def _inspect_channel_identifier(
+            self,
+            channel_identifier: typing.Optional[typing.ChannelID],
+            participant1: typing.Address,
+            participant2: typing.Address,
+            called_by_fn: str,
+    ) -> typing.ChannelID:
+        if not channel_identifier:
+            channel_identifier = self._call_and_check_result(
+                'getChannelIdentifier',
+                to_checksum_address(participant1),
+                to_checksum_address(participant2),
+            )
+        assert isinstance(channel_identifier, typing.T_ChannelID)
+        if channel_identifier == 0:
+            raise ChannelIncorrectStateError(
+                f'When calling {called_by_fn} either 0 value was given for the '
+                'channel_identifier or getChannelIdentifier returned 0, meaning '
+                f'no channel is currently open between {pex(participant1)} and '
+                f'{pex(participant2)}',
+            )
+        return channel_identifier
+
+    def channel_exists_and_not_settled(
+            self,
+            participant1: typing.Address,
+            participant2: typing.Address,
+            channel_identifier: typing.Optional[typing.ChannelID] = None,
```

Comment:
some functions have the channel_identifier as the first argument, while other as the last, can we be consistent with this?

Reply:
The `channel_identifier` is consistently last for all the functions where it's optional. If optional the latest channel identifier is requested by asking the proxy for the current channel identifier between `participant1` and `participant2`

Reply:
> The channel_identifier is consistently last for all the functions where it's optional

https://github.com/raiden-network/raiden/pull/2035/files#diff-d02574a7a7eca826412780e80a6326a3R210

Reply:
also, if the default value is `None` mypy will infer that the type is optional, so this signature is suficient:

`channel_identifier: typing.ChannelID = None`